src/bentoml/_internal/bento/build_config.py

```diff
@@ -463,7 +463,11 @@ class PythonOptions:
         default=None,
         validator=attr.validators.optional(attr.validators.instance_of(ListStr)),
     )
-    lock_packages: t.Optional[bool] = attr.field(
+    lock_packages: bool = attr.field(
```

Comment:
Should we just don't pack git packages when `lock_packages=False`? I don't think having a separate option makes sense here. 

If user don't want to pack git packages, then they shouldn't lock other non-git packages as well, and vice versa. Probably more easier to understand

Reply:
`pack_git_packages` solves multiple problems. One is `pip-tools` doesn't lock the git commit hash when locking packages, the other is to remove the need of ssh keys during `containerize`.

There are chances that users don't want to lock packages, while still don't want to inject ssh keys into containerize.